pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -400,6 +403,50 @@
     '';
   };
 
+  codesnap-nvim =
+    let
+      version = "1.3.1";
+      src = fetchFromGitHub {
+        owner = "mistricky";
+        repo = "codesnap.nvim";
+        rev = "refs/tags/v${version}";
+        hash = "sha256-nS/bAWsBQ1L4M9437Yp6FdmHoogzalKlLIAXnRZyMp0=";
+      };
+      codesnap-lib = rustPlatform.buildRustPackage {
+        pname = "codesnap-lib";
+        inherit version src;
+
+        sourceRoot = "${src.name}/generator";
+
+        cargoHash = "sha256-FTQl5WIGEf+RQKYJ4BbIE3cCeN+NYUp7VXIrpxB05tU=";
+
+        nativeBuildInputs = [
+          pkg-config
+          rustPlatform.bindgenHook
+        ];
+
+        buildInputs = [
+          libuv.dev
+        ] ++ lib.optionals stdenv.isDarwin [
+          darwin.apple_sdk.frameworks.AppKit
+        ];
+      };
+    in
+    buildVimPlugin {
+      pname = "codesnap.nvim";
+      inherit version src;
+
+      # https://github.com/mistricky/codesnap.nvim/blob/main/scripts/build_generator.sh
+      postInstall = let
+        extension = if stdenv.isDarwin then "dylib" else "so";
+      in ''
+        cp ${codesnap-lib}/lib/libgenerator.${extension} lua/generator.so
+      '';
+
+      doInstallCheck = true;
+      nvimRequireCheck = "codesnap";
```

Comment:
does it rely on an external service or do you think you could try to generate a picture in the test ?

Reply:
Is there any documentation for writing tests for Vim plugins?

Reply:
No just the code. It's not hard: nvimRequireCheck starts nvim with `:lua require'<nvimRequireCheck-value>'` while vimCommandHook starts nvim with `:vimCommandHook`

Reply:
So I was able to make the plugin work with just a single keymap:

```lua
vim.keymap.set(
  'n',
  '<leader>cs',
  ':lua require("codesnap").setup({save_path="/tmp/out.png"})<CR>:norm! ggVG<CR>:CodeSnapSave txt<CR>',
  { desc = 'Snap Code' }
)
```

Which is basically a combination of 3 commands:
1. `:lua require("codesnap").setup({save_path="/tmp/out.png"})`: setup & set out path for the snapshot
2. `:norm! ggVG`: go to top, select all the code until the bottom
3. `:CodeSnapSave txt`: snap that with document type `txt`

I'm having difficulties adding this to the `vimCommandHook`, though, which I think only checks that the command exists.

My attempt is at https://github.com/eljamm/nixpkgs/commit/0a9aa3ddb7919bc6732cfff1dd1fb448e536215c

Reply:
@eljamm wow nice ! you are trying a more elaborate test than vimCommandHook or nvimRequireCheck in which case you will need to write your own `checkPhase`. 
The test you tried to write shouldn't live in nixpkgs IMO, it should be a basic test that codesnap upstream already has and we should just run it.
 Now I can't find a testsuite in their repo (dont they have tests) ?  If you are really motivated by this you can contribute to upstream, otherwise maybe just add `nvimRequireCheck = "codesnap"`.

You can check what it does in  pkgs/applications/editors/vim/plugins/neovim-require-check-hook.sh

Reply:
> If you are really motivated by this you can contribute to upstream, otherwise maybe just add `nvimRequireCheck = "codesnap"`.

This is what is already done in this PR. Do you want me to add anything else ?

Reply:
Thanks a lot for the help. I'll check and see what I can do upstream, but for now `nvimRequireCheck` might be enough.